### PR TITLE
23283 add cd ci scripts strr pay

### PR DESCRIPTION
**Pull request by avni-work**
*Issue:*

- bcgov/entity/issues/23283

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the namex license (Apache 2.0).

### CODE REVIEW

File: .github/workflows/strr-pay-cd.yaml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        type: choice
+        options:
+          - dev
+          - test
+          - sandbox
+          - prod
+
+jobs:
+  strr-pay-cd:
+    uses: bcgov/bcregistry-sre/.github/workflows/frontend-cd.yaml@main
```
Comment (pwei1018) [inline .github/workflows/strr-pay-cd.yaml:23]: it should be backend-cd.

**Comment by pwei1018** [comment]
"devops" folder is missing. It contains vault mapping and cloud deploy file.

File: queue_services/strr-pay/devops/gcp/clouddeploy.yaml [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+      cloudsql-instances: "bcrbk9-test:northamerica-northeast1:strr-db-test"
+      service-account: "sa-api@bcrbk9-test.iam.gserviceaccount.com"
+ - targetId: bcrbk9-sandbox
+   profiles: [sandbox]
+   strategy:
+    standard:
+      verify: false
+   deployParameters:
+   - values:
+      deploy-env: "sandbox"
```
Comment (bolyachevets) [inline queue_services/strr-pay/devops/gcp/clouddeploy.yaml:55]: i think, your deploy-env values need to correspond to values in config: https://github.com/bcgov/STRR/blob/67475ec9927eec56607ba8b9d567f5b94470b55b/queue_services/strr-pay/src/strr_pay/config.py#L58 so, you have production (or staging), testing, development there, but no 'sandbox' and 'test'. So probably, 'sandbox' need to be replaced with 'staging'

File: queue_services/strr-pay/devops/gcp/clouddeploy.yaml [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+      cloudsql-instances: "bcrbk9-dev:northamerica-northeast1:strr-db-dev"
+      service-account: "sa-api@bcrbk9-dev.iam.gserviceaccount.com"
+ - targetId: bcrbk9-test
+   profiles: [test]
+   strategy:
+    standard:
+      verify: false
+   deployParameters:
+   - values:
+      deploy-env: "test"
```
Comment (bolyachevets) [inline queue_services/strr-pay/devops/gcp/clouddeploy.yaml:42]: there is no  deploy-env: "test", need to replace it with either "development" or "staging" (see my comment on sandbox env below)

File: .github/workflows/strr-pay-cd.yaml [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+        type: choice
+        options:
+          - dev
+          - test
+          - sandbox
+          - prod
+
+jobs:
+  strr-pay-cd:
+    uses: bcgov/bcregistry-sre/.github/workflows/backend-ci.yaml@main
```
Comment (pwei1018) [inline .github/workflows/strr-pay-cd.yaml:23]: typo. backend-cd

File: queue_services/strr-pay/.flake8 [hunk 5]
```diff
@@ -0,0 +1,14 @@
+[flake8]
```
Comment (pwei1018) [inline queue_services/strr-pay/.flake8:1]: flake8 configuration can be set in pyproject.yaml.

File: queue_services/strr-pay/src/strr_pay/config.py [hunk 6]
```diff
@@ -41,15 +41,15 @@
...
+from dotenv import find_dotenv
+from dotenv import load_dotenv
 
 # this will load all the envars from a .env file located in the project root (api)
 load_dotenv(find_dotenv())
 
 CONFIGURATION = {
     "development": "strr_pay.config.DevConfig",
-    "testing": "strr_pay.config.TestConfig",
+    "test": "strr_pay.config.TestConfig",
```
Comment (bolyachevets) [inline queue_services/strr-pay/src/strr_pay/config.py:52]: if you check TestConfig definition https://github.com/bcgov/STRR/blob/e7dd5533c28d9421bcf00357a5113e77a56b3371/queue_services/strr-pay/src/strr_pay/config.py#L126 it sets TESTING = True, this variable is used by Flask and changes behavior of the app, e.g., error handling. This is appropriate for unit tests, renaming 'testing' to 'test' is OK, but using this configuration for gcp 'test' profile is not a good idea.

File: queue_services/strr-pay/devops/gcp/clouddeploy.yaml [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+      service-account: "sa-api@bcrbk9-dev.iam.gserviceaccount.com"
+
+ - targetId: bcrbk9-test
+   profiles: [test]
+   strategy:
+    standard:
+      verify: false
+   deployParameters:
+   - values:
+      deploy-env: "test"
```
Comment (bolyachevets) [inline queue_services/strr-pay/devops/gcp/clouddeploy.yaml:29]: as i mentioned in the comment for renaming testing to test, you need to use an environment name that has TESTING = False. So, you have 2 choices, use 'development' here or create another config environment named 'test' (renaming current 'test' to 'unittest'), and using that instead.

File: queue_services/strr-pay/src/strr_pay/config.py (hunk 6 above)
Comment (avni-work) [reply queue_services/strr-pay/src/strr_pay/config.py]: okay studied the sample file you suggested at https://github.com/bcgov/bcros-common/blob/d9e58ac7168a4ef95953b7f760a0a777350028a3/notify-service/notify-api/src/notify_api/config.py#L177 and have made comparable changes.

**Review by pwei1018 — APPROVED** [review]

**Pull request merged**